src/AsyncWriterResult/Library.fs

```diff
@@ -27,6 +27,9 @@ module Async =
             return unwrappedF x
         }
 
+    let zip left right =
+        bind (fun l -> bind (fun r -> retn (l, r)) right) left

```

Comment:
personally think the CE is more readable in this case rather than nested inline functions. Tho maybe not so straight forward in the 'deeper' types? `logaf low`.

```suggestion
    let zip left right =
        async {
            let! l = left
            let! r = right
            return (l, r)
        }
```

Reply:
Yeah I was thinking to keep it using bind just because the zip definition should be the same in any context so if we need to implement a zip in another CE we can just copy paste this line. I've updated it to work in parallel now thought based on Don Syme's recommendation in this comment https://github.com/dotnet/fsharp/discussions/11043#discussioncomment-345815